spec/fortnox/api/types/housework_types_spec.rb

```diff
@@ -46,23 +49,135 @@
+    # it_behaves_like 'housework type', 'STORAGESELFPRODUCEDELECTRICTY', TYPE_GREEN
+    # it_behaves_like 'housework type', 'CHARGINGSTATIONELECTRICVEHICLE', TYPE_GREEN
+    # it_behaves_like 'housework type', 'OTHERCOSTS', TYPE_GREEN
+  end
+  # rubocop:enable RSpec/RepeatedDescription
+
+  it_behaves_like 'housework type', 'COOKING', TYPE_RUT, legacy: true
+  it_behaves_like 'housework type', 'TUTORING', TYPE_RUT, legacy: true
+
+  describe 'without tax reduction type' do
+    subject do
+      cassette = 'orders/housework_without_tax_reduction_type'
+      -> { VCR.use_cassette(cassette) { repository.save(document) } }
+    end
+
+    let(:document) do
+      Fortnox::API::Model::Order.new(
+        customer_number: '1',
+        order_rows: [
+          Fortnox::API::Types::OrderRow.new(
+            ordered_quantity: 1,
+            article_number: '0000',
+            housework_type: Fortnox::API::Types::HouseworkTypes['CONSTRUCTION'],
+            housework: true
+          )
+        ]
+      )
+    end
+
+    let(:error_message) do
+      'Dokument utan angiven skattereduktionstyp får inte innehålla artikelrader med husarbetestyp.'
+    end
+
+    it 'raises an error' do
+      is_expected.to raise_error(Fortnox::API::RemoteServerError, error_message)
+    end
+  end
+
+  describe 'with OTHERCOSTS' do
+    subject do
+      cassette = 'orders/housework_othercoses_invalid'
+      -> { VCR.use_cassette(cassette) { repository.save(document) } }
+    end
+
+    let(:document) do
+      Fortnox::API::Model::Order.new(
+        customer_number: '1',
+        tax_reduction_type: TYPE_ROT,
+        order_rows: [
+          Fortnox::API::Types::OrderRow.new(
+            ordered_quantity: 1,
+            article_number: '0000',
+            housework_type: Fortnox::API::Types::HouseworkTypes['OTHERCOSTS'],
+            housework: true
+          )
+        ]
+      )
+    end
+
+    let(:error_message) { 'Kan inte sätta typen övrig kostnad på en rad markerad som husarbete.' }
+
+    it "can't have housework set to true" do
+      is_expected.to raise_error(Fortnox::API::RemoteServerError, error_message)
+    end
+  end
+
+  describe 'with wrong tax reduction type' do
+    subject do
+      cassette = 'orders/housework_invalid_tax_reduction_type'
+      -> { VCR.use_cassette(cassette) { repository.save(document) } }
+    end
+
+    let(:document) do
+      Fortnox::API::Model::Order.new(
+        customer_number: '1',
+        tax_reduction_type: TYPE_RUT,
+        order_rows: [
+          Fortnox::API::Types::OrderRow.new(
+            ordered_quantity: 1,
+            article_number: '0000',
+            housework_type: Fortnox::API::Types::HouseworkTypes['CONSTRUCTION'],
+            housework: true
+          )
+        ]
+      )
+    end
+
+    let(:error_message) do
+      "Dokument med skattereduktionstypen 'rut' får inte innehålla rader med husarbetestypen 'CONSTRUCTION'."
```

Comment:
You are using a hardcoded part of the message here. If you change the test to use a different `Fortnox::API::Types::HouseworkTypes` the message will change and the test will fail. So you should probably interpolate the expected value, `CONSTRUCTION`, here.